src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -561,6 +561,13 @@ public void fetchParticipationsSubmissionsAndResultsForExam(StudentExam studentE
     public void filterParticipationForExercise(StudentExam studentExam, Exercise exercise, List<StudentParticipation> participations, boolean isAtLeastInstructor) {
         // remove the unnecessary inner course attribute
         exercise.setCourse(null);
+
+        // If test exam, filter out participations that don't belong to this student exam
+        if (studentExam.isTestExam()) {
+            List<Long> ids = studentExam.getStudentParticipations().stream().map(StudentParticipation::getId).toList();
+            participations = participations.stream().filter(participation -> ids.contains(participation.getId())).collect(Collectors.toList());
```

Comment:
Use `toList()` instead of `.collect(Collectors.toList())` ([Artemis coding conventions](https://docs.artemis.cit.tum.de/dev/guidelines/server#general-best-practices))
This will, however, change the mutability of the list. But it shouldn't have any impact on the code I believe.